pkg/podres/middleware/terminalpods/terminalpods_test.go

```diff
@@ -0,0 +1,105 @@
+package terminalpods
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+)
+
+func TestFilterFrom(t *testing.T) {
+	testCases := []struct {
+		filterPods   []*v1.Pod
+		resp         *podresourcesapi.ListPodResourcesResponse
+		expectedResp *podresourcesapi.ListPodResourcesResponse
+	}{
+		{
+			filterPods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "podA",
+						Namespace: "nsA",
+					},
+				},
+			},
+			resp: &podresourcesapi.ListPodResourcesResponse{
+				PodResources: []*podresourcesapi.PodResources{
+					{
+						Name:      "podA",
+						Namespace: "nsA",
+					},
+					{
+						Name:      "podBB",
+						Namespace: "nsA",
+					},
+					{
+						Name:      "podC",
+						Namespace: "nsAB",
+					},
+				},
+			},
+			expectedResp: &podresourcesapi.ListPodResourcesResponse{
+				PodResources: []*podresourcesapi.PodResources{
+					{
+						Name:      "podBB",
+						Namespace: "nsA",
+					},
+					{
+						Name:      "podC",
+						Namespace: "nsAB",
+					},
+				},
+			},
+		},
+		{
+			filterPods: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "podA",
+						Namespace: "nsA",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+				},
+			},
+			resp: &podresourcesapi.ListPodResourcesResponse{
+				PodResources: []*podresourcesapi.PodResources{
+					{
+						Name:      "foo",
+						Namespace: "bar",
+					},
+					{
+						Name:      "bar",
+						Namespace: "foo",
+					},
+					{
+						Name:      "podA",
+						Namespace: "nsA",
+					},
+				},
+			},
+			expectedResp: &podresourcesapi.ListPodResourcesResponse{
+				PodResources: []*podresourcesapi.PodResources{
+					{
+						Name:      "bar",
+						Namespace: "foo",
+					},
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		FilterFrom(tc.resp, tc.filterPods)
```

Comment:
this works. If you have time, please either:
1. document and enforce a ordering (any given ordering) in `FilterFrom`, because now it seems implicit/assumed
2. sort values before to compare them (or add a one-line comment here documenting `cmp.Diff` does the sorting - if it does...)

Reply:
Done